storages/backends/gcloud.py

```diff
@@ -191,9 +190,9 @@ def delete(self, name):
     def exists(self, name):
         if not name:  # root element aka the bucket
             try:
-                self.bucket
+                self.client.get_bucket(self.bucket)
                 return True
-            except ImproperlyConfigured:
+            except (ImproperlyConfigured, NotFound):
```

Comment:
ImproperlyConfigured can no longer be caught here, I think.
```suggestion
            except NotFound:
```